src/components/tabs/tabs.component.js

```diff
@@ -234,11 +234,14 @@ const Tabs = ({
         return summaryOfMessages.map((value) => `• ${value}`).join("\n");
       };
 
-      const tabTitle = (
+      return (
         <TabTitle
           position={isInSidebar ? "left" : position}
           className={child.props.className || ""}
           dataTabId={tabId}
+          data-role={
+            titleProps !== undefined ? titleProps["data-role"] : undefined
+          }
```

Comment:
since we're limiting `titleProps` to just contain `data-role` (at least for now), I would much prefer to spread props: `{...titleProps}` - I'm pretty sure you can spread undefined as an object and it will add no properties.

I especially think it's right to do it this way since I can only assume the point of having a `titleProps` object with just a single property is to expand this interface later if needed, and we don't want to have to add all those potential props individually.

Reply:
Agree with Robin would prefer to see it spread here for simplicity's sake

Reply:
Originally I decided to do this so we weren't inadvertently supporting any prop that could be passed via `titleProps`, while still using the `titleProps` approach in case we need to expose more props later. However, I can change this if you think this is excessive and would improve readability

Reply:
I see your point - thanks for explaining. On balance I'd rather spread them, if consumers try to pass something else they will hopefully be aware it's undocumented and is being used at their own risk so I don't see it as a problem. But I won't argue any further if you decide to keep it as you had it.

Reply:
although even with it this way I would prefer to see it changed to `data-role={titleProps?.["data-role"]}`, which is more readable and should come to the same thing.

Reply:
Yeah I'd prefer either this or destructure (requires default value of {}) it from `titleProps` as I think someone may come across your implementation in the future and refactor it and miss your valid reasoning you mention above @Parsium 

```
const {dataRole} = titleProps;
```

Reply:
I agree with both of you on this - will push a change now 👍🏼